feature-libs/my-account/organization/components/order-approval/list/order-approval-list.component.ts

```diff
@@ -0,0 +1,79 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import {
+  EntitiesModel,
+  RoutingService,
+  TranslationService,
+} from '@spartacus/core';
+import {
+  B2BSearchConfig,
+  OrderApproval,
+  OrderApprovalService,
+} from '@spartacus/my-account/organization/core';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-order-approval-list',
+  templateUrl: './order-approval-list.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OrderApprovalListComponent {
+  constructor(
+    protected routing: RoutingService,
+    protected orderApprovalService: OrderApprovalService,
+    protected translation: TranslationService
+  ) {
+    this.fetchApprovalListPage({});
+  }
+
+  protected PAGE_SIZE = 5;
+  sortType: string;
+
+  orderApprovals$: Observable<EntitiesModel<OrderApproval>>;
+
+  changeSortCode(sortCode: string): void {
+    const fetchParams: B2BSearchConfig = {
+      sort: sortCode,
+      currentPage: 0,
+    };
+    this.sortType = sortCode;
+    this.fetchApprovalListPage(fetchParams);
+  }
+
+  pageChange(page: number): void {
+    const fetchParams: B2BSearchConfig = {
+      sort: this.sortType,
+      currentPage: page,
+    };
+    this.fetchApprovalListPage(fetchParams);
+  }
+
+  getSortLabels(): Observable<{ byDate: string; byOrderNumber: string }> {
+    return combineLatest([
```

Comment:
Performance:
This observable is re-created on each change detection cycle. We could change it into a property.

Reply:
Updated